x/ante/ante.go

```diff
@@ -49,14 +49,16 @@ func NewLogMsgDecorator(cdc codec.Codec) LogMsgDecorator {
 
 // AnteHandle logs all messages in blocks
 func (d LogMsgDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, next sdk.AnteHandler) (sdk.Context, error) {
-	msgs := tx.GetMsgs()
-
-	for _, msg := range msgs {
-		logger(ctx).Debug(fmt.Sprintf("received message of type %s in block %d: %s",
-			proto.MessageName(msg),
-			ctx.BlockHeight(),
-			string(d.cdc.MustMarshalJSON(msg)),
-		))
+	if !simulate {
```

Comment:
if simulate { return...} to avoid unnecessary nested if blocks

Reply:
That would duplicate the `next` call at the end though

Reply:
which is fine

Reply:
Code duplication seems worse compared to a bit more nesting

Reply:
I strongly disagree, especially in golang. Do you wanna start seeing
```
    if err := ...; err == nil {
        if err := ...; err == nil {
            if err := ...; err == nil {
                ...
            }
        }
    }
```

Reply:
What exactly makes you think this kind of duplicate is bad? What makes ante handler special in this case?

Reply:
nested-ifs compared to early exit creates the very problem that when someone reads it, he has to figure out what happens if `simulate` is `true` by scrolling down the entire block to just find out nothing is gonna happen. It's confusing and unreadable. When one sees `if simulate { return ... }`, it's clear immediately that this function isn't gonna do anything when `simultate` is set to `true` and also makes a lot of sense since logging shouldn't happen if in the simulation mode.

Reply:
Updated it. I get the benefit of early returns on failure cases. But in this case, it's not a failure, so requires duplication of logic. My concern is if someone changes the logic of `next` at the end, they need to remember to do this for the early exit too. This is specifically concerning for ante handle checks since they perform some authentication and are critical. The vote module had a duplication related bug like this if I recall. But anyways, up to you, since I haven't coded in Go much.

Reply:
They wouldn't need to remember to do anything to the early exit. The whole point here is that we wanna make sure nothing should ever happen in the logging ante handler if simuate is true. If they have to do something to the early exit, I can see a lot of stuff being questionable about what that person is doing.

Reply:
ante handler isn't anything special. It's just like middlewares in express.js or interceptors in gRPC.